src/Command/Pull/PullCommandBase.php

```diff
@@ -569,6 +576,9 @@ protected function promptChooseDatabases(
       $chosen_database_keys = $this->io->askQuestion($question);
       $chosen_databases = [];
       if (count($chosen_database_keys) === 1 && $chosen_database_keys[0] === 'all') {
+        if (count($environment_databases) > 10) {
+          $this->io->warning('You have chosen to pull down more than 10 databases. This could exhaust your disk space.');
```

Comment:
We cannot do this in Cloud IDE until we're checking for the available disk space prior.